pyaerocom/griddeddata.py

```diff
@@ -2021,16 +2021,19 @@ def aerocom_savename(self, data_id=None, var_name=None,
         """Get filename for saving following AeroCom conventions"""
         from pyaerocom.io.helpers import aerocom_savename
         if vert_code is None:
-            try:
-                from pyaerocom.io.fileconventions import FileConventionRead
-                f = self.from_files[0]
-                fconv = FileConventionRead().from_file(f)
-                vert_code = fconv.get_info_from_file(f)['vert_code']
-            except Exception:
-                pass
+            if self.metadata['vert_code'] is not None:
+                vert_code=self.metadata['vert_code']
+            else:
+                try:
+                    from pyaerocom.io.fileconventions import FileConventionRead
+                    f = self.from_files[0]
+                    fconv = FileConventionRead().from_file(f)
+                    vert_code = fconv.get_info_from_file(f)['vert_code']
+                except Exception:
+                    pass
```

Comment:
Maybe write someting to logger before passing?

Reply:
Good point. I think this method tries to be too smart, I'll revise in #456